bioptim/dynamics/lagrange_interpolation.py

```diff
@@ -0,0 +1,252 @@
+                \\frac{1}{(\\tau_j - \\tau_k)}
+                \\prod_{\\substack{r \\neq j, r \\neq k}}^{} \\frac{(\\tau - \\tau_r)}{(\\tau_j - \\tau_r)}
+              \\right],
+
+        which, when factored appropriately, yields the compact form above.
+        """
+        sum_term = 0
+        for k in range(self.polynomial_order):
+            if k == j:
+                continue
+
+            partial_Ljk = self.partial_lagrange_polynomial(j, time_control_interval, k)
+            sum_term += 1.0 / (self.time_grid[j] - self.time_grid[k]) * partial_Ljk
+
+        return sum_term
+
+    def interpolate(self, y_values: list[MX | SX | float], time_control_interval: MX | SX | float) -> MX | SX:
+        """
+        Compute the Lagrange interpolation of a set of values at the given time.
+
+        Parameters
+        ----------
+        y_values : list of MX or SX
+            The values to interpolate at each collocation point.
+        time_control_interval : MX or SX
+            The symbolic variable (CasADi `MX` or `SX`) representing
+            the dimensionless time within the control interval.
+
+        Returns
+        -------
+        MX or SX
+            A CasADi symbolic expression representing the Lagrange interpolation
+            of the given values at the time ``time_control_interval``.
+        """
+        self._check_y_values(y_values)
+        interpolated_value = 0
+        for j in range(self.polynomial_order):
+            interpolated_value += y_values[j] * self.lagrange_polynomial(j, time_control_interval)
+        return interpolated_value
+
+    def interpolate_first_derivative(
+        self, y_values: list[MX | SX | float], time_control_interval: MX | SX | float
+    ) -> MX | SX:
+        """
+        Compute the first derivative of the Lagrange interpolation of a set of values at the given time.
+
+        Parameters
+        ----------
+        y_values : list of MX or SX
+            The values to interpolate at each collocation point.
+        time_control_interval : MX or SX
+            The symbolic variable (CasADi `MX` or `SX`) representing
+            the dimensionless time within the control interval.
+
+        Returns
+        -------
+        MX or SX
+            A CasADi symbolic expression representing the first derivative
+            of the Lagrange interpolation of the given values at the time ``time_control_interval``.
+        """
+        self._check_y_values(y_values)
+        interpolated_value = 0
+        for j in range(self.polynomial_order):
+            interpolated_value += y_values[j] * self.lagrange_polynomial_derivative(j, time_control_interval)
+        return interpolated_value
+
+    def _check_y_values(self, y_values):
+        if len(y_values) != self.polynomial_order:
+            raise ValueError(
+                f"Length of y_values ({len(y_values)}) must match the polynomial order ({self.polynomial_order})"
+            )
+
+
+def main():
```

Comment:
I am guessing this will be moved to a test or an example ?

Reply:
I was planning to keep it here. As it it is a straightforward use of the source code. 

But yeah it's a good sign because this class is easily testable itself ;)